docs/architecture/adr-001-reports-type-field-expansion.md

```diff
@@ -0,0 +1,128 @@
+# ADR 001: Reports type field review
+
+## Changelog
+
+- September 9th, 2021: Initial draft
+- September 14th, 2021: First proposal
+
+## Status
+
+PROPOSED
+
+## Abstract
+
+Report's `type` field SHOULD be expanded and enhanced to make possible adding 
+more reasons and elaborate more why a post has been reported. To standardise the different `reasons`
+inside the network and across all the social networks there will be a group of default ones. These can be
+extended by governance proposal later on.
+
+## Context
+
+Centralized social networks give users the possibility to select multiple reasons when they're about to report a post.
+By doing this, the report become itself more explicit and complete and will later help the moderator/community to judge and choose the best
+way to handle it. 
+Actually inside Desmos we don't handle the reporting system like described. Instead, we rely on our `Report` object 
+which contains the `type` field. The `type` field takes care to display the reason behind the report, but at the moment,
+it can only store 1 reason at time. Despite this can be handy and quick during a first implementation attempt, it
+may be not enough for future decentralized social networks and social enabled apps.  
+
+Users need an exhaustive way to report contents on decentralized platforms, so having a repeated `type` field that allows
+to do this should be the better option to improve the whole report system. This ADR proposes such changes inside the current
+`Report` type.
+
+## Decision
+
+We will replace the `Report` `type` field with a more explicit `reasons` type.  
+As briefly described before,
+the `reasons` will be an array of strings that specify the reasons why the post has been reported.
+
+```protobuf
+// Report is the struct of a post's reports
+message Report {
+  option (gogoproto.equal) = true;
+  option (gogoproto.goproto_stringer) = true;
+  
+  // ID of the post for which the report has been created
+  string post_id = 1 [
+    (gogoproto.customname) = "PostID",
+    (gogoproto.jsontag) = "post_id",
+     (gogoproto.moretags) = "yaml:\"post_id\""
+   ];
+  
+   // Identifies the reasons of the report
+   repeated string reasons = 2 [ 
+     (gogoproto.jsontag) = "reasons", 
+     (gogoproto.moretags) = "yaml:\"reasons\"" 
+   ];
+
+   // User message
+   string message = 3 [
+    (gogoproto.jsontag) = "message",
+    (gogoproto.moretags) = "yaml:\"message\""
+  ];
+   
+  // Identifies the reporting user
+   string user = 4 [ 
+     (gogoproto.jsontag) = "user", 
+     (gogoproto.moretags) = "yaml:\"user\"" 
+   ];
+ }
+```
+
+To standardise reasons across all the network, the available ones will be included as a 
+`posts` module parameter that can be extended in the future or completely changed via governance
+proposal.
```

Comment:
Personally, I have thought a lot about this approach I don't think it's the best one for two main reasons: 

1. Not all subspaces might want to have the same reporting types. 
    Think about at Facebook and Twitter: they have two completely different ways of handling explicit content. Facebook bans that kind of content immediately, while Twitter allows it to exist on their platform.
2. In the long run, as new subspaces are created to accommodate different applications, the refresh of this list might take too long for such subspaces to exist. Having to wait even one day might be sufficient for developers to desist in adding their report type. Also, each time a new report type is added for everyone, all developers must make sure they are supporting it inside their application, which can be extremely difficult and time consuming. 

For these reasons, I think there is a better way of handling reports types: store them inside each subspace.

If we allow developers to freely creating their own custom reports type, we will make it more easy for them to only handle significant reports. Also, we would not tie them to the overall protocol governance for a thing that is so connected to the kind of application you are developing. Finally, having the valid report types list inside each subspace allows developers to not support reports at all: they simply have to set an empty list of supported reports types. This can be handy for some applications that do not want to deal with reports at all. 

I also propose to have the default supported types of reports empty for each subspace, so that each developer is forced to specify them one by one.

Reply:
This makes sense to me. While reviewing the proposal I've also thought that, in order to simplify and ease the complexity of later checks on these, the subspace id can be added to the report type. So we will have: 

```protobuf
// Report is the struct of a post's reports
message Report {
  option (gogoproto.equal) = true;
  option (gogoproto.goproto_stringer) = true;
  
  // ID of the post for which the report has been created
  string post_id = 1 [
    (gogoproto.customname) = "PostID",
    (gogoproto.jsontag) = "post_id",
    (gogoproto.moretags) = "yaml:\"post_id\""
   ];
  
   // Identifies the reasons of the report
   repeated string reasons = 2 [ 
     (gogoproto.jsontag) = "reasons", 
     (gogoproto.moretags) = "yaml:\"reasons\"" 
   ];

   // User message
   string message = 3 [
    (gogoproto.jsontag) = "message",
    (gogoproto.moretags) = "yaml:\"message\""
  ];
   
  // Identifies the reporting user
   string user = 4 [ 
     (gogoproto.jsontag) = "user", 
     (gogoproto.moretags) = "yaml:\"user\"" 
   ];
 }
```